dbms/src/Flash/Management/ManualCompact.cpp

```diff
@@ -0,0 +1,220 @@
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Common/setThreadName.h>
+#include <Flash/Management/ManualCompact.h>
+#include <Flash/ServiceUtils.h>
+#include <Storages/DeltaMerge/RowKeyRange.h>
+#include <Storages/StorageDeltaMerge.h>
+#include <Storages/Transaction/TMTContext.h>
+
+#include <ext/scope_guard.h>
+#include <future>
+
+namespace DB
+{
+
+namespace Management
+{
+
+ManualCompactManager::ManualCompactManager(const Context & db_context_)
+    : db_context(db_context_)
```

Comment:
Not sure whether db_context is ok (or using a global_context better?)

Reply:
This should be a reference to "global_context" as ManualCompactManager lives through the lifetime of the TiFlash process.
"db_context" usually indicates its lifetime is only for one query session